app/monitoringapi.go

```diff
@@ -0,0 +1,122 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package app
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	eth2client "github.com/attestantio/go-eth2-client"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/lifecycle"
+	"github.com/obolnetwork/charon/cluster"
+	"github.com/obolnetwork/charon/p2p"
+)
+
+// newReadyHandler returns a http.HandlerFunc which returns 200 when both the beacon node is synced and all quorum peers can be pinged  in parallel within a timeout. Returns 500 otherwise.
+func newReadyHandler(ctx context.Context, conf Config, life *lifecycle.Manager, tcpNode host.Host, lock cluster.Lock) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		eth2Svc, _, err := newETH2Client(ctx, conf, life, nil)
+		if err != nil {
+			writeResponse(w, http.StatusInternalServerError, "Couldn't initialize ETH2 client")
+		}
+
+		syncing, err := beaconNodeSyncing(ctx, eth2Svc)
+		if err != nil {
+			writeResponse(w, http.StatusInternalServerError, "Failed to get beacon sync state")
+		}
+
+		if syncing {
+			writeResponse(w, http.StatusInternalServerError, "Beacon node not synced")
+			return
+		}
+
+		peers, err := lock.Peers()
```

Comment:
pass in peers

Reply:
does that mean initializing `eth2Cl` and `peers` in the `wireMonitoringAPI` function and then passing them to the `newReadyHandler` function?

Reply:
done